### PR TITLE
rename test function for datasets

### PR DESCRIPTION
**Pull request by sandalns**
### Linked issue(s):
Fixes KOL-3939
### What change does this PR introduce and why?
`test` for datasets is just a result upload, renaming it to make more sense
### Please check if the PR fulfills these requirements

- [ ] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [ ] Relevant docs have been added / updated

### CODE REVIEW

File: docs/reference/dataset/upload-results.md [hunk 1]
```diff
@@ -0,0 +1 @@
+::: kolena.dataset.upload_results
```
Comment (ly-kolena) [inline docs/reference/dataset/upload-results.md:1]: also need to update places where this file is referenced

File: mkdocs.yml [hunk 2]
```diff
@@ -53,7 +53,7 @@ nav:
           - <code>fetch_dataset_history</code>: reference/dataset/fetch-dataset-history.md
           - <code>fetch_dataset</code>: reference/dataset/fetch-dataset.md
           - <code>fetch_results</code>: reference/dataset/fetch-results.md
-          - <code>test</code>: reference/dataset/test.md
+          - <code>test</code>: reference/dataset/upload-results.md
```
Comment (ly-kolena) [inline mkdocs.yml]: test -> upload-results

File: kolena/dataset/__init__.py [hunk 3]
```diff
@@ -16,12 +16,15 @@
 from kolena.dataset.dataset import fetch_dataset
 from kolena.dataset.dataset import register_dataset
 from kolena.dataset.evaluation import fetch_results
-from kolena.dataset.evaluation import test
+from kolena.dataset.evaluation import upload_results
+from kolena.workflow import annotation, asset
```
Comment (ly-kolena) [inline kolena/dataset/__init__.py]: need to remove this (have you rebased against latest trunk)?

File: kolena/dataset/__init__.py [hunk 4]
```diff
@@ -16,12 +16,15 @@
...
 
 __all__ = [
     "register_dataset",
     "fetch_dataset_history",
     "fetch_dataset",
     "fetch_results",
-    "test",
+    "upload_results",
+    "annotation",
```
Comment (ly-kolena) [inline kolena/dataset/__init__.py]: need to remove these 2

File: kolena/dataset/__init__.py (hunk 3 above)
Comment (sandalns) [reply kolena/dataset/__init__.py]: I took the wrong diff when I fixed the merge conflict earlier, good catch

File: examples/dataset/keypoint_detection/keypoint_detection/upload_results.py [hunk 5]
```diff
@@ -37,7 +37,8 @@ def run(args: Namespace) -> None:
         results.append(dict(locator=record.locator, raw_bboxes=bboxes, raw_faces=faces, **metrics))
 
     df_results = pd.DataFrame.from_records(results)
-    test(args.dataset, args.model, df_results)
+    upload_results(args.dataset, args.model, df_results)
+    return 0
```
Comment (ly-kolena) [inline examples/dataset/keypoint_detection/keypoint_detection/upload_results.py]: `return 0` is not needed

**Comment by ly-kolena** [comment]
Non-blocking: Is there a backend PR follow up coming up?

**Review by ly-kolena — APPROVED** [review]

**Review by ly-kolena — APPROVED** [review]

**Review by munkyshi — APPROVED** [review]

**Comment by ly-kolena** [comment]
cc @sandalns latest trunk should fix the example failure, can you merge/rebase and merge this after https://github.com/kolenaIO/kolena/pull/424?

**Pull request merged**